src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -368,8 +368,8 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
             {
                 if (isRMW)
                 {
-                    assert((targetReg == op1Reg) || (targetReg != op2Reg));
-                    assert((targetReg == op1Reg) || (targetReg != op3Reg));
+                    assert((targetReg == op1Reg) || (targetReg != op2Reg) || genIsSameLocalVar(intrin.op1, intrin.op2));
```

Comment:
There are several other assertions like this in `hwintrinsiccodegenarm64.cpp` that I suspect need to be updated. Should we update those proactively? I suppose if they fail because they're overzealous, that wouldn't affect the correctness of product builds.